hooks/infinite/useInfiniteMessagesById.ts

```diff
@@ -0,0 +1,29 @@
+import { useInfiniteQuery } from '@tanstack/react-query';
+
+import { getMessagesById } from '~/apis/message';
+
+const SIZE = 6;
+
+const useInfiniteMessagesById = (partnerId: number) => {
+  return useInfiniteQuery({
+    queryKey: ['messages', partnerId],
+    queryFn: ({ pageParam }) =>
+      getMessagesById({
+        partnerId,
+        size: SIZE,
+        cursorId: pageParam || undefined,
+      }),
+    initialPageParam: 0,
+    getNextPageParam: lastMessage => {
+      const { messageResponses } = lastMessage;
+      const { length } = messageResponses;
+
+      const lastId = length ? messageResponses[length - 1].id : undefined;
+
+      return lastMessage.hasNext ? lastId : undefined;
+    },
+    select: data => data.pages.flatMap(page => page.messageResponses),
```

Comment:
혹시 축약해서 사용해볼수도 있을까요?

```suggestion
    select: ({pages})=> pages.flatMap(({messageResponses})=> messageResponses),
```

Reply:
오호 넵 알겠습니다!! 중복 줄여보겠습니다